### DISCUSSION TITLE
Quando não é vantajoso utilizar a plataforma WordPress?

### DISCUSSION THREAD

**Discussion by leonardosouzafaria** [frontendbr/forum#500] [Discussão]
Hoje me surgiu uma duvida bem simples, quais são os pontos negativos do WordPress, quando não usar?
Estive estudando sobre a plataforma e sempre os sites criam artigos falando sobre benefícios de se utilizar WordPress, etc, e nunca apontam quando não usar. 
E minha dúvida é essa... Quando não usar o WordPress em seu projeto?

**Comment by leticiacostadias** [comment]
Quando o projeto for crescer "demais". Leia-se quando você vai fazer algo maior que um site institucional ou um blog. Quanto mais regras você coloca no WP, mais difícil vai ser pra ele organizar isso tudo sem criar uma bola de neve.

**Comment by ghost** [comment]
WordPress é uma ótimo cms, trabalho para uma agencia de Orlando e lá só se usa ele.
É de praxe falarem mal, dizerem que fica desorganizado, pesado, entre outras coisas. Porém isso acontece com tudo no meio do desenvolvimento. Depende de quem desenvolve e como desenvolve.

Acontece também que ele pode ser gerenciado por pessoas sem muito ou nenhum conhecimento em dev, então a quantidade de plugins será enorme.

Outro ponto é a quantidade de materiais, artigos e dicas que vc encontra a respeito. O que provavelmente não terá, caso opte por algum outro novo no mercado.

Respondendo a sua pergunta, se tratando de um site institucional que vá ter um gerenciamento de conteúdo como profissionais, serviços, fotos, entre outras coisas que um site comumente tem, ele é uma boa opção.

**Comment by woliveiras** [comment]
Concordo um pouco com a @leticiacostadias e bastante com o @wdss93.

O problema do WordPress é o mesmo problema de qualquer outra ferramenta: os usuários (nós desenvolvedores, no caso).

Quando não temos conhecimento suficiente, fazemos coisa errada.

As pessoas tem o costume de achar que o WordPress é tão fácil que não precisa ser estudado antes de se fazer algo grande... Não é bem assim.

É uma excelente ferramenta, usamos no trampo. Nosso principal problema é o tema, que foi um tema comprado, que veio todo cheio de código amarrado com arame e pra dar manutenção nisso é bem chato.

Qualquer alteração é um risco de vida.

Se tivesse sido criado pensado em crescer, esse Blog seria muito mais fácil de manter.

Hoje em dia da pra usar o WordPress até se quiser fazer o seu conteúdo aparecer em uma SPA, então quase não tem casos onde não se use ele.

- Se quiser criar do zero: https://1fix.io/blog/2014/11/05/angularjs-json-api-wp-theme/
- Se quiser partir de um ponto quase pronto: https://github.com/frontpressorg/frontpress

Um caso que vejo é se você não quer gastar com hospedagem, mas acho difícil alguém querer fazer um site e não planejar esse custo.

**Comment by rubenmarcus** [comment]
Muito legal essa Issue, estou pensando em fazer um tema wordpress com vue ou react, ou os dois, já tinha visto alguns em Angular, Especialmente porque eu preciso refazer meu portfolio e quero ter a liberdade de edita-lo se precisar em qualquer situação , ex:  estar sem computador, de uma lanhouse, de um hotel, de qualquer lugar que tenha acesso a internet.

Aqui na Printi usamos o Bolt.
Ouvi pessoal falar bem do Grav.. e do Ghost. Pra falar a verdade eu queria fazer algo em Nodejs, Mas.. como fiz um contrato de 3 anos com a Bluehost por um plano bem em conta, vai ter que ser algo em PHP mesmo.

**Uma dúvida:**

É possivel pegar um json que você já usa como api pra um site e porta-lo para posts no wordpress, ou qualquer cms que seja ? 

Ex:
https://github.com/rubenmarcus/old-portfolio/blob/master/data/portfolio-br.json



Eu já tenho meu portfolio meio que estruturado, queria fazer isso.

Pra falar a verdade eu queria mesmo fazer muito algo em Node.JS (Meteor), Mongodb e etc pra aprender essas tecnologias , por no curriculo e não só ficar lendo artigo sobre, Mas o fato de eu ter um servidor PHP por 3 anos, pesa bastante, de eu não ter mais um custo com Node.. se bem que tem servidores gratuitos de node tb nao é?

**Comment by mtmr0x** [comment]
Essa issue não poderia ter `id` melhor do que `500`. Obrigado por isso, primeiro de tudo. E em segundo e mais importante, eu diria que eu concordo muito com esse post que diz por que nao usar o Wordpress: https://www.webbedfeet.uk/news/10-reasons-why-web-developers-should-not-use-wordpress

Se qualquer um desses items poderia ser um problema para seu projeto, então não recomendaria usar.

**Comment by fdaciuk** [comment]
Eu vejo duas desvantagens em usar WP:
1. Quando você não sabe programar em PHP;
2. Quando você não conhece a API do WP, e a forma como ele funciona.

Sabendo isso, você escala como quiser. Um bom exemplo? O [wordpress.com](https://wordpress.com) é feito usando WordPress e usa **uma única** base de dados MySQL. [Aqui](https://wordpress.com/activity/) você encontra alguns números que podem te ajudar a entender se o WP escala bem ou não =)

**Comment by woliveiras** [comment]
@mtmr0x relaxo essa observação do 500 ai! 😂  😂  😂  😂  😂  😂

**Comment by woliveiras** [comment]
Nosso blog também é com o WordPress: https://www.vivadecora.com.br/revista/

**`Está cheio de problemas, mas vamos resolver com o novo tema, viu`**

O 500 era comum até organizarmos os servidores e serviços da AWS. 😬 😬 😬

**Comment by alanhr** [comment]
Cara WordPress na minha opinião é o pior CMS PHP que existe por conta da forma que o código foi feito.
Hoje temos October é grav que na minha opinião são os melhores.

**Comment by nandomoreirame** [comment]
nossa que hater @alanhr :rofl:

**Comment by mariovalney** [comment]
**Quando não usar o WordPress em seu projeto?**

Quando não for um site/sistema **voltado a conteúdo**. Não adianta querer usar um Gerenciador de Conteúdo para fazer algo que não seja gerenciar conteúdo. **Mas muita coisa é conteúdo**: e-commerce, rede social, site institucional, plataforma de classificados, etc... 

-------------------------------

No mais:

- **E segurança?**
Com a quantidade de **atualizações** e gente trabalhando no Core bugs e **falhas tem mais chances de serem corrigidas**, diferente de algum framework/cms que não recebe manutenção constante. Além disso, é servidor e os temas/plugins (código seu ou de terceiro).

- **E escalabilidade?**
WordPress.com gerencia milhões de blogs com mais de **409 milhões de usuários** visualizando mais de 15.5 bilhões de páginas ( além de **41.7 milhões de novos posts** e 60.5 milhões de novos comentários) por mês. (Acho que seu site com 10.000 usuários e 1.000.000 de page views vai dar certo se fizer direito)

- **E o código do Core?**
Seu cliente não quer saber do código do core... **faça o Tema e seus Plugins direito** e vai ter toda a performance e segurança (e "beleza" de código) que importa para a aplicação dele.

**Comment by alanhr** [comment]
@nandomoreirame  cara não fui hater :), apenas expressei minha sincera opinião sobre o WP.

**Comment by allysonsouza** [comment]
Para quem não tem ideia da escalabilidade, segurança, estabilidade, e etc. do WordPress, além do [WordPress.com](wordpress.com) citado pelo @fdaciuk, deixo aqui essa pequena lista: https://www.codeinwp.com/blog/who-uses-wordpress/

**Comment by ghost** [comment]
Ótimas indagações aqui, me ajudaram bastante...eu to pensando em fazer um site de notícias para um nicho junto com podcasts e vídeos no youtube, acham que valeria fazer via jekyll ou no wordpress mesmo? Valeu

**Comment by mariovalney** [comment]
WordPress.

**Comment by mksanti** [comment]
WordPress.

**Comment by renanlara** [comment]
Eu já participei de alguns projetos grandes em WordPress e vi muito a equipe que estava participando do projeto meio que falar mal do CMS por conta da altitude em que ele chegou. A @leticiacostadias sabe bem de quem estou falando... Mas, antes de tudo, devemos levar em conta que todo o problema que deve ter ocorrido com o projeto poderia ter sido resolvido muito antes. Quem trabalha em agência sabe que o atendimento promete coisas para o cliente em prazos surreais sem saber da equipe de desenvolvimento o que dar para fazer com a melhor forma de fazer e dentro de um prazo justo. O que não acontece.

O resultado é: O desenvolvedor vai desenvolver coisas em horas, o que poderia demorar dias para desenvolver. O dev vai meter GoHorse até nas narinas do projeto e a consequência é um sistema lento, inseguro, cheio de bugs e por aí vai como iria acontecer **com _qualquer_ outro sistema.**
